src/Symfony/Component/HttpClient/MockHttpClient.php

```diff
@@ -45,6 +48,10 @@ public function __construct($responseFactory = null, string $baseUri = null)
             })();
         }
 
+        if ($defaultOptions) {
+            [, $this->defaultOptions] = self::prepareRequest(null, null, $defaultOptions, self::OPTIONS_DEFAULTS);
```

Comment:
we should pass `true` as last argument to allow passing any options

Reply:
Thanks, fixed.